### DISCUSSION TITLE
Snowpack warning: import.meta is not available

### DISCUSSION THREAD

**Discussion by rmrbytes** [gildas-lormeau/zip.js#251] [Q&A]
I am placing this `warning` that showed up only as a matter or record. I have not found any problem in the use of the library under the Snowpack environment (*I needed to test my project in another bundle environment because webpack was giving me some problems*).

The warning showed up during the build process of Snowpack:

```
> _snowpack/pkg/@zipjs/zipjs.js: warning: "import.meta" is not available in the configured target environment and will be empty
    4843 │     workerData.worker = new Worker(new URL(workerData.scripts[0], import.meta.url));
         ╵                                                                   ~~~~~~~~~~~
```
 Thanks

**Comment by gildas-lormeau** [comment]
Thank you for the information. I guess this is the same issue than https://github.com/snowpackjs/snowpack/discussions/2513.

The fix would be to set `target` to `es2020` in your config file.

**Comment by shadow-light** [comment]
This seems to break in Webpack 4 too

```
 ERROR  Failed to compile with 1 error                                                                10:11:20 AM

 error  in ./node_modules/@zip.js/zip.js/lib/core/codecs/codec-pool-worker.js

Module parse failed: Unexpected token (82:70)
You may need an appropriate loader to handle this file type, currently no loaders are configured to process this file. See https://webpack.js.org/concepts#loaders
| 	let messageTask;
| 	if (!workerData.interface) {
> 		workerData.worker = new Worker(new URL(workerData.scripts[0], import.meta.url));
| 		workerData.worker.addEventListener(MESSAGE_EVENT_TYPE, onMessage, false);
| 		workerData.interface = {
```

**Comment by gildas-lormeau** [comment]
@shadow-light Did you try to install https://www.npmjs.com/package/@open-wc/webpack-import-meta-loader? see https://github.com/gildas-lormeau/zip.js/issues/212#issuecomment-769737609

**Comment by shadow-light** [comment]
Thanks for suggestion. I just imported `'@zip.js/zip.js/dist/zip'` instead to save time.
